npm install does not work

Installing homebridge-dht-sensor via `sudo npm install -g homebridge-dht-sensor` fails for me on Raspberry Pi, running latest Rasbian `Linux raspi 4.4.26+ #915 Thu Oct 20 17:02:14 BST 2016 armv6l GNU/Linux` (Raspbian Jessie Lite) with user `pi`. Node version `v4.4.5`.

Do I need some special dependencies, what am I missing?

```
pi@raspi:~ $ sudo npm install -g homebridge-dht-sensor
|
> node-dht-sensor@0.0.5 install /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor
> node-gyp rebuild

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/4.4.5"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp"
make: Entering directory '/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/build'
  CXX(target) Release/obj.target/node-dht-sensor/node-dht-sensor.o
../node-dht-sensor.cpp:191:26: error: ‘Arguments’ does not name a type
 Handle<Value> Read(const Arguments& args) {
                          ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h: In function ‘v8::Handle<v8::Value> Read(const int&)’:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:885:13: error: ‘v8::HandleScope::HandleScope()’ is protected
   V8_INLINE HandleScope() {}
             ^
../node-dht-sensor.cpp:192:17: error: within this context
     HandleScope scope;
                 ^
../node-dht-sensor.cpp:202:41: error: no matching function for call to ‘v8::Object::New()’
     Local<Object> readout = Object::New();
                                         ^
../node-dht-sensor.cpp:202:41: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2933:24: note: static v8::Local<v8::Object> v8::Object::New(v8::Isolate*)
   static Local<Object> New(Isolate* isolate);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2933:24: note:   candidate expects 1 argument, 0 provided
../node-dht-sensor.cpp:203:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("humidity"), Number::New(humidity));
                  ^
../node-dht-sensor.cpp:203:69: error: no matching function for call to ‘v8::Number::New(float&)’
     readout->Set(String::NewSymbol("humidity"), Number::New(humidity));
                                                                     ^
../node-dht-sensor.cpp:203:69: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:204:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("temperature"), Number::New(temperature));
                  ^
../node-dht-sensor.cpp:204:75: error: no matching function for call to ‘v8::Number::New(float&)’
     readout->Set(String::NewSymbol("temperature"), Number::New(temperature));
                                                                           ^
../node-dht-sensor.cpp:204:75: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:205:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("isValid"), Boolean::New(result == 0));
                  ^
../node-dht-sensor.cpp:205:72: error: no matching function for call to ‘v8::Boolean::New(bool)’
     readout->Set(String::NewSymbol("isValid"), Boolean::New(result == 0));
                                                                        ^
../node-dht-sensor.cpp:205:72: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note: static v8::Local<v8::Boolean> v8::Boolean::New(v8::Isolate*, bool)
 Local<Boolean> Boolean::New(Isolate* isolate, bool value) {
                ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:206:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("errors"), Number::New(2 - retry));
                  ^
../node-dht-sensor.cpp:206:68: error: no matching function for call to ‘v8::Number::New(int)’
     readout->Set(String::NewSymbol("errors"), Number::New(2 - retry));
                                                                    ^
../node-dht-sensor.cpp:206:68: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:207:18: error: ‘class v8::HandleScope’ has no member named ‘Close’
     return scope.Close(readout);
                  ^
../node-dht-sensor.cpp: At global scope:
../node-dht-sensor.cpp:210:30: error: ‘Arguments’ does not name a type
 Handle<Value> ReadSpec(const Arguments& args) {
                              ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h: In function ‘v8::Handle<v8::Value> ReadSpec(const int&)’:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:885:13: error: ‘v8::HandleScope::HandleScope()’ is protected
   V8_INLINE HandleScope() {}
             ^
../node-dht-sensor.cpp:211:17: error: within this context
     HandleScope scope;
                 ^
../node-dht-sensor.cpp:213:14: error: request for member ‘Length’ in ‘args’, which is of non-class type ‘const int’
     if (args.Length() < 2) {
              ^
../node-dht-sensor.cpp:214:44: error: ‘New’ is not a member of ‘v8::String’
        ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
                                            ^
../node-dht-sensor.cpp:214:85: error: ‘ThrowException’ was not declared in this scope
        ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
                                                                                     ^
../node-dht-sensor.cpp:215:21: error: ‘class v8::HandleScope’ has no member named ‘Close’
        return scope.Close(Undefined());
                     ^
../node-dht-sensor.cpp:215:37: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
        return scope.Close(Undefined());
                                     ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:218:28: error: invalid types ‘const int[int]’ for array subscript
     int sensorType = args[0]->Uint32Value();
                            ^
../node-dht-sensor.cpp:220:45: error: ‘New’ is not a member of ‘v8::String’
         ThrowException(Exception::TypeError(String::New("Specified sensor type is invalid.")));
                                             ^
../node-dht-sensor.cpp:220:94: error: ‘ThrowException’ was not declared in this scope
         ThrowException(Exception::TypeError(String::New("Specified sensor type is invalid.")));
                                                                                              ^
../node-dht-sensor.cpp:221:22: error: ‘class v8::HandleScope’ has no member named ‘Close’
         return scope.Close(Undefined());
                      ^
../node-dht-sensor.cpp:221:38: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
         return scope.Close(Undefined());
                                      ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:227:49: error: ‘New’ is not a member of ‘v8::String’
             ThrowException(Exception::TypeError(String::New("Failed to initialize.")));
                                                 ^
../node-dht-sensor.cpp:227:86: error: ‘ThrowException’ was not declared in this scope
             ThrowException(Exception::TypeError(String::New("Failed to initialize.")));
                                                                                      ^
../node-dht-sensor.cpp:228:26: error: ‘class v8::HandleScope’ has no member named ‘Close’
             return scope.Close(Undefined());
                          ^
../node-dht-sensor.cpp:228:42: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
             return scope.Close(Undefined());
                                          ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:232:22: error: invalid types ‘const int[int]’ for array subscript
     int gpio = args[1]->Uint32Value();
                      ^
../node-dht-sensor.cpp:241:41: error: no matching function for call to ‘v8::Object::New()’
     Local<Object> readout = Object::New();
                                         ^
../node-dht-sensor.cpp:241:41: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2933:24: note: static v8::Local<v8::Object> v8::Object::New(v8::Isolate*)
   static Local<Object> New(Isolate* isolate);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2933:24: note:   candidate expects 1 argument, 0 provided
../node-dht-sensor.cpp:242:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("humidity"), Number::New(humidity));
                  ^
../node-dht-sensor.cpp:242:69: error: no matching function for call to ‘v8::Number::New(float&)’
     readout->Set(String::NewSymbol("humidity"), Number::New(humidity));
                                                                     ^
../node-dht-sensor.cpp:242:69: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:243:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("temperature"), Number::New(temperature));
                  ^
../node-dht-sensor.cpp:243:75: error: no matching function for call to ‘v8::Number::New(float&)’
     readout->Set(String::NewSymbol("temperature"), Number::New(temperature));
                                                                           ^
../node-dht-sensor.cpp:243:75: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:244:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("isValid"), Boolean::New(result == 0));
                  ^
../node-dht-sensor.cpp:244:72: error: no matching function for call to ‘v8::Boolean::New(bool)’
     readout->Set(String::NewSymbol("isValid"), Boolean::New(result == 0));
                                                                        ^
../node-dht-sensor.cpp:244:72: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note: static v8::Local<v8::Boolean> v8::Boolean::New(v8::Isolate*, bool)
 Local<Boolean> Boolean::New(Isolate* isolate, bool value) {
                ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:245:18: error: ‘NewSymbol’ is not a member of ‘v8::String’
     readout->Set(String::NewSymbol("errors"), Number::New(2 - retry));
                  ^
../node-dht-sensor.cpp:245:68: error: no matching function for call to ‘v8::Number::New(int)’
     readout->Set(String::NewSymbol("errors"), Number::New(2 - retry));
                                                                    ^
../node-dht-sensor.cpp:245:68: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note: static v8::Local<v8::Number> v8::Number::New(v8::Isolate*, double)
   static Local<Number> New(Isolate* isolate, double value);
                        ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:2486:24: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp:246:18: error: ‘class v8::HandleScope’ has no member named ‘Close’
     return scope.Close(readout);
                  ^
../node-dht-sensor.cpp: At global scope:
../node-dht-sensor.cpp:249:32: error: ‘Arguments’ does not name a type
 Handle<Value> Initialize(const Arguments& args) {
                                ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h: In function ‘v8::Handle<v8::Value> Initialize(const int&)’:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:885:13: error: ‘v8::HandleScope::HandleScope()’ is protected
   V8_INLINE HandleScope() {}
             ^
../node-dht-sensor.cpp:250:17: error: within this context
     HandleScope scope;
                 ^
../node-dht-sensor.cpp:252:14: error: request for member ‘Length’ in ‘args’, which is of non-class type ‘const int’
     if (args.Length() < 2) {
              ^
../node-dht-sensor.cpp:253:45: error: ‘New’ is not a member of ‘v8::String’
         ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
                                             ^
../node-dht-sensor.cpp:253:86: error: ‘ThrowException’ was not declared in this scope
         ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
                                                                                      ^
../node-dht-sensor.cpp:254:22: error: ‘class v8::HandleScope’ has no member named ‘Close’
         return scope.Close(Undefined());
                      ^
../node-dht-sensor.cpp:254:38: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
         return scope.Close(Undefined());
                                      ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:257:16: error: invalid types ‘const int[int]’ for array subscript
     if (!args[0]->IsNumber() || !args[1]->IsNumber()) {
                ^
../node-dht-sensor.cpp:257:40: error: invalid types ‘const int[int]’ for array subscript
     if (!args[0]->IsNumber() || !args[1]->IsNumber()) {
                                        ^
../node-dht-sensor.cpp:258:45: error: ‘New’ is not a member of ‘v8::String’
         ThrowException(Exception::TypeError(String::New("Wrong arguments")));
                                             ^
../node-dht-sensor.cpp:258:76: error: ‘ThrowException’ was not declared in this scope
         ThrowException(Exception::TypeError(String::New("Wrong arguments")));
                                                                            ^
../node-dht-sensor.cpp:259:22: error: ‘class v8::HandleScope’ has no member named ‘Close’
         return scope.Close(Undefined());
                      ^
../node-dht-sensor.cpp:259:38: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
         return scope.Close(Undefined());
                                      ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:262:28: error: invalid types ‘const int[int]’ for array subscript
     int sensorType = args[0]->Uint32Value();
                            ^
../node-dht-sensor.cpp:264:45: error: ‘New’ is not a member of ‘v8::String’
         ThrowException(Exception::TypeError(String::New("Specified sensor type is not supported")));
                                             ^
../node-dht-sensor.cpp:264:99: error: ‘ThrowException’ was not declared in this scope
         ThrowException(Exception::TypeError(String::New("Specified sensor type is not supported")));
                                                                                                   ^
../node-dht-sensor.cpp:265:22: error: ‘class v8::HandleScope’ has no member named ‘Close’
         return scope.Close(Undefined());
                      ^
../node-dht-sensor.cpp:265:38: error: too few arguments to function ‘v8::Local<v8::Primitive> v8::Undefined(v8::Isolate*)’
         return scope.Close(Undefined());
                                      ^
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:315:27: note: declared here
   friend Local<Primitive> Undefined(Isolate* isolate);
                           ^
../node-dht-sensor.cpp:270:22: error: invalid types ‘const int[int]’ for array subscript
     GPIOPort = args[1]->Uint32Value();
                      ^
../node-dht-sensor.cpp:272:18: error: ‘class v8::HandleScope’ has no member named ‘Close’
     return scope.Close(v8::Boolean::New(initialize() == 0));
                  ^
../node-dht-sensor.cpp:272:58: error: no matching function for call to ‘v8::Boolean::New(bool)’
     return scope.Close(v8::Boolean::New(initialize() == 0));
                                                          ^
../node-dht-sensor.cpp:272:58: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note: static v8::Local<v8::Boolean> v8::Boolean::New(v8::Isolate*, bool)
 Local<Boolean> Boolean::New(Isolate* isolate, bool value) {
                ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:7603:16: note:   candidate expects 2 arguments, 1 provided
../node-dht-sensor.cpp: In function ‘void RegisterModule(v8::Handle<v8::Object>)’:
../node-dht-sensor.cpp:276:17: error: ‘NewSymbol’ is not a member of ‘v8::String’
     target->Set(String::NewSymbol("read"),
                 ^
../node-dht-sensor.cpp:277:43: error: no matching function for call to ‘v8::FunctionTemplate::New(v8::Handle<v8::Value> (&)(const int&))’
                 FunctionTemplate::New(Read)->GetFunction());
                                           ^
../node-dht-sensor.cpp:277:43: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note: static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::Isolate*, v8::FunctionCallback, v8::Local<v8::Value>, v8::Local<v8::Signature>, int)
   static Local<FunctionTemplate> New(
                                  ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note:   no known conversion for argument 1 from ‘v8::Handle<v8::Value>(const int&) {aka v8::Local<v8::Value>(const int&)}’ to ‘v8::Isolate*’
../node-dht-sensor.cpp:278:17: error: ‘NewSymbol’ is not a member of ‘v8::String’
     target->Set(String::NewSymbol("readSpec"),
                 ^
../node-dht-sensor.cpp:279:47: error: no matching function for call to ‘v8::FunctionTemplate::New(v8::Handle<v8::Value> (&)(const int&))’
                 FunctionTemplate::New(ReadSpec)->GetFunction());
                                               ^
../node-dht-sensor.cpp:279:47: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note: static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::Isolate*, v8::FunctionCallback, v8::Local<v8::Value>, v8::Local<v8::Signature>, int)
   static Local<FunctionTemplate> New(
                                  ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note:   no known conversion for argument 1 from ‘v8::Handle<v8::Value>(const int&) {aka v8::Local<v8::Value>(const int&)}’ to ‘v8::Isolate*’
../node-dht-sensor.cpp:280:17: error: ‘NewSymbol’ is not a member of ‘v8::String’
     target->Set(String::NewSymbol("initialize"),
                 ^
../node-dht-sensor.cpp:281:49: error: no matching function for call to ‘v8::FunctionTemplate::New(v8::Handle<v8::Value> (&)(const int&))’
                 FunctionTemplate::New(Initialize)->GetFunction());
                                                 ^
../node-dht-sensor.cpp:281:49: note: candidate is:
In file included from /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/node.h:42:0,
                 from ../node-dht-sensor.cpp:1:
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note: static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::Isolate*, v8::FunctionCallback, v8::Local<v8::Value>, v8::Local<v8::Signature>, int)
   static Local<FunctionTemplate> New(
                                  ^
/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/.node-gyp/4.4.5/include/node/v8.h:4349:34: note:   no known conversion for argument 1 from ‘v8::Handle<v8::Value>(const int&) {aka v8::Local<v8::Value>(const int&)}’ to ‘v8::Isolate*’
../node-dht-sensor.cpp: In function ‘v8::Handle<v8::Value> Read(const int&)’:
../node-dht-sensor.cpp:208:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
../node-dht-sensor.cpp: In function ‘v8::Handle<v8::Value> ReadSpec(const int&)’:
../node-dht-sensor.cpp:247:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
../node-dht-sensor.cpp: In function ‘v8::Handle<v8::Value> Initialize(const int&)’:
../node-dht-sensor.cpp:273:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
node-dht-sensor.target.mk:86: recipe for target 'Release/obj.target/node-dht-sensor/node-dht-sensor.o' failed
make: *** [Release/obj.target/node-dht-sensor/node-dht-sensor.o] Error 1
make: Leaving directory '/usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
 System Linux 4.4.26+
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/homebridge-dht-sensor/node_modules/node-dht-sensor
gyp ERR! node -v v4.4.5
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
```

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
   },
   "dependencies": {
     "babel-core": "6.14.0",
-    "node-dht-sensor": "git+https://github.com/sweetpi/node-dht-sensor.git#master"
+    "node-dht-sensor": "0.0.29"
   },
   "devDependencies": {
     "babel-cli": "6.14.0",

--- a/src/createDhtSensor.js
+++ b/src/createDhtSensor.js
@@ -26,22 +26,23 @@ export default function createDhtSensor({ Service, Characteristic }) {
     }
 
     getValue(what, callback) {
-      const result = sensorLib.readSpec(this.type, this.pin)
-      if (result.humidity !== 0 && result.temperature !== 0) {
-        this.humidity = result.humidity
-        this.temperature = result.temperature
-        this.lastUpdate = Date.now()
-      }
+      sensorLib.read(this.type, this.pin, (err, temperature = 0, humidity = 0) => {
+        if (!err && humidity !== 0 && temperature !== 0) {
+          this.humidity = humidity
+          this.temperature = temperature
+          this.lastUpdate = Date.now()
+        }
 
-      if (Date.now() - this.lastUpdate >= this.cacheTimeout) {
-        return callback(new Error('cannot get sensor data'), null)
-      }
+        if (Date.now() - this.lastUpdate >= this.cacheTimeout) {
+          return callback(new Error('cannot get sensor data'), null)
+        }
 
-      switch (what) {
-        case 'temperature': return callback(null, this.temperature)
-        case 'humidity': return callback(null, this.humidity)
-        default: return callback(new Error(`cannot get unknown property '${what}'`), null)
-      }
+        switch (what) {
+          case 'temperature': return callback(null, this.temperature)
+          case 'humidity': return callback(null, this.humidity)
+          default: return callback(new Error(`cannot get unknown property '${what}'`), null)
+        }
+      })
     }
 
     getServices() {

